software/omni_motion.py

```diff
@@ -0,0 +1,148 @@
+    # Pyserial stuff
+    def close(self):
+        self.ser.close()
+
+    # Big math, returns speed of a wheel in mainboard units
+    def calculate_wheel_speed(self, motor_num, robot_speed, robot_angle, speed_rot):
+        wheel_linear_velocity = robot_speed * math.cos(robot_angle - self.wheel_angles[motor_num - 1]) + self.wheel_distance_from_center * speed_rot
+        wheel_angular_speed_mainboard_units = wheel_linear_velocity * self.wheel_speed_to_mainboard_units
+        return wheel_angular_speed_mainboard_units
+
+    # x - sideways, positive to the right
+    # y - forward, positive to the front
+    # r - rotation, positive anticlockwise
+    # While moving it also rotates the thrower with self.thrower_speed
+    def move(self, speed_x, speed_y, speed_r, thrower_speed=0):        
+        robot_angle = math.atan2(speed_y, speed_x)
+        robot_speed = math.sqrt(math.pow(speed_x, 2) + math.pow(speed_y, 2))
+        #print("Angle:", robot_angle)
+        #print("Speed:", robot_speed)
+        if abs(robot_speed) > self.max_speed or abs(speed_r) > self.max_speed:
+            LOGE("Speed to large")
+            return
+
+        #M1 front left
+        m1 = int(self.calculate_wheel_speed(1, robot_speed, robot_angle, speed_r))
+        #M2 front right
+        m2 = int(self.calculate_wheel_speed(2, robot_speed, robot_angle, speed_r))
+        #M3 back
+        m3 = int(self.calculate_wheel_speed(3, robot_speed, robot_angle, speed_r))
+
+        self.send_data(m1, m2, m3, thrower_speed)
+        #print("Cmd:", m1, m2, m3)
+        #print("Actual:", self.receive_data())
+
+    # Orbit around something with constant linear (sideways) speed, cur values allow for adjustments in speed, to make it more precise
+    # speed - positive value starts orbiting anticlockwise (robot moves right)
+    # radius, cur_radius in millimeters
+    # cur_object_x - x coordinate of the center of the orbital trajectory
+    def orbit(self, radius, speed_x, cur_radius, cur_object_x):
+        speed_y = 0
+        speed_r = 1000 * speed_x / radius
+
+        # Correct radius check
+        if cur_radius > 600:
+            LOGE("Invalid radius, radius: " + str(cur_radius))
+            return
+        # Radius adjustment
+        if cur_radius > (radius + self.buffer_x) or cur_radius < (radius - self.buffer_x):
+            speed_y -= (radius - cur_radius) / 100 * self.y_const
+        # Centering object, rotational speed adjustment
+        if cur_object_x > (self.middle_x + self.buffer_x) or cur_object_x < (self.middle_x - self.buffer_x):
+            speed_r += (self.middle_x - cur_object_x) / 100 * self.r_const
+
+        #print("x:", speed_x, "y:", speed_y, "r:", speed_r, "cur_rad:", cur_radius, "cur_x:", cur_object_x)
+        self.move(speed_x, speed_y, speed_r)
+        self.prev_rad = cur_radius
+
+
+    # Throw ball with given strength
+    # Discrete call not continuous, use within a loop
+    # Value range 48 - 2047
+    def throw(self, strength):

```

Comment:
Moved the strength calculation into mainboard.py (previously named omni_motion.py).
Implemented throw_raw() method so that thrower can still be controlled by raw motor values if needed during testing.